code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -371,51 +391,67 @@
-		if((C.stat) || (C.handcuffed))
+	var/judgement_criteria = judgement_criteria()
+	for(var/mob/living/carbon/nearby_carbons in view(7,src)) //Let's find us a criminal
+		if((nearby_carbons.stat) || (nearby_carbons.handcuffed))
 			continue
 
-		if((C.name == oldtarget_name) && (world.time < last_found + 100))
+		if((nearby_carbons.name == oldtarget_name) && (world.time < last_found + 100))
 			continue
 
-		threatlevel = C.assess_threat(judgment_criteria, weaponcheck=CALLBACK(src, PROC_REF(check_for_weapons)))
+		threatlevel = nearby_carbons.assess_threat(judgement_criteria, weaponcheck = CALLBACK(src, PROC_REF(check_for_weapons)))
 
 		if(!threatlevel)
 			continue
 
 		else if(threatlevel >= 4)
-			target = C
-			oldtarget_name = C.name
+			target = nearby_carbons
+			oldtarget_name = nearby_carbons.name
 			speak("Level [threatlevel] infraction alert!")
-			playsound(loc, pick('sound/voice/beepsky/criminal.ogg', 'sound/voice/beepsky/justice.ogg', 'sound/voice/beepsky/freeze.ogg'), 50, FALSE)
-			visible_message("<b>[src]</b> points at [C.name]!")
+			if(bot_type == ADVANCED_SEC_BOT)
+				playsound(src, pick('sound/voice/ed209_20sec.ogg', 'sound/voice/edplaceholder.ogg'), 50, FALSE)
+			else
+				playsound(src, pick('sound/voice/beepsky/criminal.ogg', 'sound/voice/beepsky/justice.ogg', 'sound/voice/beepsky/freeze.ogg'), 50, FALSE)
+			visible_message("<b>[src]</b> points at [nearby_carbons.name]!")
 			mode = BOT_HUNT
 			INVOKE_ASYNC(src, PROC_REF(handle_automated_action))
 			break
-		else
-			continue
 
 /mob/living/simple_animal/bot/secbot/proc/check_for_weapons(var/obj/item/slot_item)
 	if(slot_item && (slot_item.item_flags & NEEDS_PERMIT))
 		return TRUE
 	return FALSE
 
 /mob/living/simple_animal/bot/secbot/explode()
+
 	visible_message("<span class='boldannounce'>[src] blows apart!</span>")
 	var/atom/Tsec = drop_location()
-
-	var/obj/item/bot_assembly/secbot/Sa = new (Tsec)
-	Sa.build_step = 1
-	Sa.add_overlay("hs_hole")
-	Sa.created_name = name
-	new /obj/item/assembly/prox_sensor(Tsec)
-	if(!noloot)
+	if(bot_type == ADVANCED_SEC_BOT)
+		var/obj/item/bot_assembly/ed209/ed_assembly = new(Tsec)
+		ed_assembly.build_step = ASSEMBLY_FIRST_STEP
+		ed_assembly.add_overlay("hs_hole")
+		ed_assembly.created_name = name
+		new /obj/item/assembly/prox_sensor(Tsec)
+		var/obj/item/gun/energy/disabler/disabler_gun = new(Tsec)
+		disabler_gun.cell.charge = 0
+		disabler_gun.update_appearance()
+		if(prob(50))
+			new /obj/item/bodypart/l_leg/robot(Tsec)
+			if(prob(25))
+				new /obj/item/bodypart/r_leg/robot(Tsec)
+		if(prob(25))//50% chance for a helmet OR vest
+			if(prob(50))
+				new /obj/item/clothing/head/helmet(Tsec)
+			else
+				new /obj/item/clothing/suit/armor/vest(Tsec)
+	else
+		var/obj/item/bot_assembly/secbot/secbot_assembly = new(Tsec)
+		secbot_assembly.build_step = ASSEMBLY_FIRST_STEP
+		secbot_assembly.add_overlay("hs_hole")
+		secbot_assembly.created_name = name
+		new /obj/item/assembly/prox_sensor(Tsec)
```

Comment:
Right now this proc is exactly like the code you're aiming to fix with this PR, and I question why code very specific to ED209, not potential future ranged bots, is being defined in the parent type. 

If component parts are stored in a list, most of this proc could be simplified to this to make it drop an assortment of random parts

```
for(var/obj/item/I in component_parts)
    if(prob(50))
        drop_part(I, loc)
```

And this also future-proofs it against later bots. 


> Alright, I think ive cleaned it to a reasonable degree that the ranged var is no longer an issue.

Er.. no this looks exactly like it did before. You're hard-spawning items that may have nothing to do with what was used in the construction of the ED209. Use a list or use the variables you removed at bare minimum.

Reply:
Might be an idea to have a unit test which constructs an EDD from a non-standard vest type, explodes it and then confirms that the correct construction types are being dropped.